plugins/modules/system/homectl.py

```diff
@@ -0,0 +1,618 @@
+            ],
+            "status": {
+                "e9ed2a5b0033427286b228e97c1e8343": {
+                    "diskCeiling": 21845405696,
+                    "diskFloor": 268435456,
+                    "diskSize": 3221225472,
+                    "service": "io.systemd.Home",
+                    "signedLocally": true,
+                    "state": "inactive"
+                }
+             },
+            "userName": "james",
+        }
+    }
+'''
+
+import crypt
+import json
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import jsonify
+from ansible.module_utils.common.text.formatters import human_to_bytes
+
+
+class Homectl(object):
+    '''#TODO DOC STRINGS'''
+
+    def __init__(self, module):
+        self.module = module
+        self.state = module.params['state']
+        self.name = module.params['name']
+        self.password = module.params['password']
+        self.storage = module.params['storage']
+        self.disksize = module.params['disksize']
+        self.resize = module.params['resize']
+        self.realname = module.params['realname']
+        self.realm = module.params['realm']
+        self.email = module.params['email']
+        self.location = module.params['location']
+        self.iconname = module.params['iconname']
+        self.homedir = module.params['homedir']
+        self.imagepath = module.params['imagepath']
+        self.uid = module.params['uid']
+        self.gid = module.params['gid']
+        self.umask = module.params['umask']
+        self.memberof = module.params['memberof']
+        self.skeleton = module.params['skeleton']
+        self.shell = module.params['shell']
+        self.environment = module.params['environment']
+        self.timezone = module.params['timezone']
+        self.locked = module.params['locked']
+        self.passwordhint = module.params['passwordhint']
+        self.sshkeys = module.params['sshkeys']
+        self.language = module.params['language']
+        self.notbefore = module.params['notbefore']
+        self.notafter = module.params['notafter']
+        self.mountopts = module.params['mountopts']
+
+        self.result = {}
+
+    # Cannot run homectl commands if service is not active
+    def homed_service_active(self):
+        is_active = True
+        cmd = ["systemctl", "show", "systemd-homed.service", "-p", "ActiveState"]
+        rc, show_service_stdout, stderr = self.module.run_command(cmd)
+        if rc == 0:
+            state = show_service_stdout.rsplit("=")[1]
+            if state.strip() != "active":
+                is_active = False
+        return is_active
+
+    def user_exists(self):
+        # Get user properties if they exist in json
+        exists = False
+        cmd = [self.module.get_bin_path('homectl', True)]
+        cmd.append("inspect")
+        cmd.append(self.name)
+        cmd.append("-j")
+        rc, stdout, stderr = self.module.run_command(cmd, use_unsafe_shell=True)
```

Comment:
What is `use_unsafe_shell=True` for here and in the other calls? `use_unsafe_shell=True` should only be used if absolutely necessary.

Reply:
ok, will change these and test.